oxidation/crates/parsec_client_types/src/local_manifest.rs

```diff
@@ -324,6 +324,24 @@ impl LocalFileManifest {
         reference.version = remote_manifest.version;
         reference == *remote_manifest
     }
+
+    pub fn evolve_and_mark_updated(
+        mut self,
+        size: u64,
+        blocks: Vec<Vec<Chunk>>,
+        timestamp: DateTime,
+    ) -> Self {
+        self.need_sync = true;
+        self.size = size;
+        self.blocks = blocks;
+        self.updated = timestamp;
+        self
+    }
+
+    pub fn evolve_single_block(mut self, block: u64, new_chunk: Chunk) -> Self {
```

Comment:
Why builder pattern ? Do you want to chain something ?

Reply:
Not really, I just wanted to be consistent with the other `evolve_*` methods implemented in this module. 

I guess that the reason why it's been implemented this way is because the original python methods are creating new instances for each `evolve`. Here, the instance is mutated and returned instead, probably to avoid the cost of copying? I don't mind either way.

Reply:
This pattern can be used for an update to give you ideas:
```Rust
let file_manifest_updated = file_manifest.set_need_sync(true)
  .set_size(size)
  .set_blocks(blocks)
  .set_updated(timestamp)
  .set_single_block(block, new_chunk)
```

Others pattern are:
```Rust
fn set_field(&mut self, field)
```
and
```Rust
let file_manifest_updated = FileManifest {
  field1: field1,
  field2: field2,
  ..file_manifest
};
```

(Ps: Change block type to usize, because it's what the function need)